boards/arm/96b_carbon/Kconfig.defconfig

```diff
@@ -10,12 +10,19 @@ if BOARD_96B_CARBON
 config BOARD
 	default 96b_carbon
 
-if BT
+if SPI
 
-config SPI
+config SPI_2
 	default y
 
-config SPI_STM32
+config SPI_2_IRQ_PRI
```

Comment:
Isn't this more of an application specific setting than a board setting?

Reply:
It is, but we already define I2C and UART IRQ priority in SOC's dtsi files. Moreover, we specify a default SPI_*_IRQ_PRI value for other boards in the same way (e.g. hexiwear_k64, em_starterkit). We can override these default values at application level using a dt overlay file or prj.conf.

Reply:
I didn't know about that; thanks for the pointers. I still think board level is the wrong place to put this, though.

I could maybe see arguments to do it at SoC level (though on ARM, I think using the default priority ordering determined by the NVIC makes more sense).

Do you know of a design rationale for why it's done elsewhere? I'm not sure it makes sense to copy what's done elsewhere without good reason here.

Reply:
I just did it to follow the same approach as other boards, until I push a PR for STM32 devices to get SPI configuration from DT (like I2C and USB).

Reply:
IMO it'd be better to drop this from the board files. Otherwise when the values in board files start to drift, it could be a source of confusing bugs when porting an application between boards. But I don't think this should block the merge since it can be overridden.

Reply:
I removed SPI_*_IRQ_PRI (I left it only on carbon because I wanted to run the SPI test cases on it) from the board files. I used test case filtering to disable some tests when the peripheral is not supported. I also push a PR to enable STM32 SPI devices to get their configuration from DT.